remote-policy/src/main/java/com/google/apigee/RemotePolicyExecution.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apigee;
+
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Decodes an Execution Protocol Buffer Message and sets a flow variable before returning to caller.
+ * Designed to be deployed as a Google Cloud Function.
+ */
+public class RemotePolicyExecution implements HttpFunction {
+
+  /**
+   * Reads and decodes the Execution Protocol Buffer Message and sets a flow variable and returns
+   * the Execution Message back to caller.
+   *
+   * @param request contents of HTTP request containing the Protocol Buffer Message bytes
+   * @param response HttpResponse sent in response to Http request containing modified Protocol
+   *     Buffer Message with new flow variable set
+   * @throws IOException
```

Comment:
Does this still throw this exception if throwables are caught?

Reply:
There's actually another IOException potentially thrown from the `response.getOutputStream().write(execution.toByteArray());` line within the catch block. If that exception is thrown then we won't be able to write anything in the response back to the caller, and I believe the exception is bubbled up to Cloud Functions to be inspected in the console.